viewer/viewer-awt/src/main/java/org/icepdf/ri/common/utility/search/SearchToolBar.java

```diff
@@ -15,46 +15,45 @@
  */
 package org.icepdf.ri.common.utility.search;
 
-import org.icepdf.ri.common.DropDownButton;
 import org.icepdf.ri.common.SwingController;
 
 import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.text.MessageFormat;
 import java.util.ResourceBundle;
-import java.util.prefs.Preferences;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
 
-import static org.icepdf.ri.util.PropertiesManager.*;
-
-public class SearchToolBar extends JToolBar implements ActionListener {
+public class SearchToolBar extends JToolBar implements ActionListener, BaseSearchComponent {
```

Comment:
There is a small bug with how the enabled/disable state is set from the SwingController.  This component needs to make sure the child elements are also enabled/disable as document are open/closed.  Currently you can click on any menu item or buttons and corresponding exceptions are thrown.